mapper/tedge_mapper/src/sm_c8y_mapper/tests.rs

```diff
@@ -319,6 +319,123 @@ async fn mapper_publishes_software_update_status_and_software_list_onto_c8y_topi
     assert!(received_software_list);
 }
 
+#[tokio::test]
+#[cfg_attr(not(feature = "mosquitto-available"), ignore)]
+#[serial]
+async fn mapper_fails_during_sw_update_recovers_and_process_response() {
```

Comment:
This test is definitely better than the other one written in python. Not because of the PL, but because how we ensure the race condition is reproduced.

Reply:
I agree, to be more deterministic, now all the requests to the tedge_agent are answered by the local subscribers. Basically, the tedge_agent is mocked here.